pyccel/codegen/printing/cwrappercode.py

```diff
@@ -273,88 +268,187 @@ def get_cast_function_call(self, cast_type, arg):
 
         return FunctionCall(cast_function, [arg])
 
+    # --------------------------------------------------------------------
+    # Functions that take care of collecting (data type, rank) checks and creating the error code
+    # --------------------------------------------------------------------
+    def _generate_TypeError_message(self, variable):
+        """
+        Generate TypeError message from the variable informations (datatype, precision)
+        """
+        dtype     = variable.dtype
+
+        if isinstance(dtype, NativeBool):
+            precision = ''
+        if isinstance(dtype, NativeComplex):
+            precision = '{} bit '.format(variable.precision * 2 * 8)
+        else:
+            precision = '{} bit '.format(variable.precision * 8)
+
+        message = '"{var_name} must be {precision}{dtype}"'.format(
+                        var_name  = variable,
+                        precision = precision,
+                        dtype     = self._print(variable.dtype))
+        return message
+
+    def _get_array_type_check(self, variable, collect_var):
+        """
+        Responsible for collecting array type check and creating the
```

Comment:
Grammar : `Responsible of` -> `Responsible for`

This error appears a few times

Reply:
forgive my poor english

Reply:
no problem

Reply:
done 0da2e5b

Reply:
I wondered why you corrected this in the wrong direction. Somehow I managed to put it on the wrong line. "Responsible for" is correct english

Reply:
-_-

Reply:
sorry

Reply:
np xD
3260790